### ISSUE TITLE
The docstrings for `prefixedMethods` and `accumulateMethods` need some polish

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5909]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5909|
|Type|enhancement|
|Created|2012-08-23 19:00:04Z|
|Branch|https://github.com/twisted/twisted/tree/reflect-docs-5909|

A few issues:

  1. `prefixedMethods` is only documented via a sentence fragment
  1. `prefixedMethods` has no parameter or (clear) return value documentation
  1. `accumulateMethods` duplicates its signature in its docstring
  1. `accumulateMethods` used the deprecated first-person convention
  1. `accumulateMethods` has no parameter documentation

Attachments:

* [patch (bug 5909).txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/be9/be90b3688766d214d2a9e5a71405b91d02ffea69/1cf93fe522bc8307b9c68f86eb31e6c5e7107dc6.txt) (1170 bytes) - added by gehrhorn on 2012-09-26 00:45:55Z - patch file for bug #5909

<details><summary>Searchable metadata</summary>

```
trac-id__5909 5909
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_reflect_docs_5909 branches/reflect-docs-5909
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation_easy documentation easy
time__1345748404000000 1345748404000000
changetime__1356566025000000 1356566025000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

You should probably also document curClass argument is private.

**Comment by twisted-trac** [comment]
|<img alt="gehrhorn's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|gehrhorn set status to `closed`|
|-|-|

I improved the documentation for prefiedMethods and accumulateMethods and attached a patch.

**Comment by twisted-trac** [comment]
|<img alt="gehrhorn's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|gehrhorn set status to `reopened`|
|-|-|

I set the status to "fixed", which I don't thing I was supposed to do.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince set owner to gehrhorn<br>@tomprince set status to `new`|
|-|-|

This patch deals with points 1, 3 and 4.

The arguments and return value should have explicit documentation (with ```@param``` and ```@return```). See the example [here](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html#auto9). There are also several examples in the file.

Having added explicit argument and return documentation, the description of the behavior could perhaps be simplified (it seems to me that the wording is more complex than necessary in order to describe the arguments inline), but this is optional.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Which patch?

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince commented|
|-|-|

The attached patch, that was submitted for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [36632]) Branching to 'reflect-docs-5909'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [36633]) Apply patch%20%28bug%205909%29.txt 

refs [#5909](https://github.com/twisted/twisted/issues/5909)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun removed owner|
|-|-|

I applied the patch to the linked branch and made some further doc improvements to it.  [Latest twistedchecker build](http://buildbot.twistedmatrix.com/builders/twistedchecker/builds/50) and [latest documentation build](http://buildbot.twistedmatrix.com/builders/documentation/builds/2776).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_11"></a>|@tomprince set owner to @exarkun|
|-|-|

Please commit after fixing points 2+3 (and optionally 1).

  1. ```prefixedMethodNames``` only refers to a specific class, but ```addMethodNamesToDict``` refers to the class and all its base classes, though  one is implemented in terms of the other. Also, ```addMethodNamesToDict``` looks like an implementation detail (I've filed [#6227](https://github.com/twisted/twisted/issues/6227) to deprecate it).
  1. ```addMethodNamesToDict``` doesn't return anything.
  1. ```accumulateMethodNames``` almost always wants to be called as 
```
methodsDict = {}
reflect.accumulateMethods(object, methodsDict, '<prefix>_')
# call methods in methodsDict
```
     But it isn't clear from the documentation that that is how it should be called. An example would be helpful here.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

Thanks.  Fixed points 1 and 2 in r36654 and r36655 respectively.  I think point 3 is scope creep and I'm inclined not to bother with it for this ticket (these are crappy APIs, anyway, that hardly no one ever cares about or uses).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [36656]) Merge reflect-docs-5909

Author: gehrhorn, exarkun
Reviewer: tom.prince
Fixes: [#5909](https://github.com/twisted/twisted/issues/5909)

Bring the API documentation for some `twisted.python.reflect` functions up
to modern standards.
